pallets/account/src/tests.rs

```diff
@@ -0,0 +1,248 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(test)]
+
+use super::*;
+use frame_support::{assert_noop, assert_ok};
+use mock::*;
+
+use ethers::{
+    contract::{Eip712, EthAbiType},
+    core::types::{transaction::eip712::Eip712, Bytes},
+};
+use parity_scale_codec::Encode;
+use sp_runtime::{traits::StaticLookup, AccountId32, MultiAddress};
+
+/// EIP712 Payload struct
+#[derive(Eip712, EthAbiType, Clone)]
+#[eip712(
+        name = "Astar EVM Claim",
+        version = "1",
+        chain_id = 1024,
+        // mock genisis hash
+        raw_salt = "0x4545454545454545454545454545454545454545454545454545454545454545"
+    )]
+struct Claim {
+    substrate_address: Bytes,
+}
+
+fn claim_signature(who: &AccountId32, secret: &libsecp256k1::SecretKey) -> [u8; 65] {
```

Comment:
Nothing is _claimed_ in this function - at least to me that's confusing down below in tests.

Reply:
I'll rename it to - `get_evm_signature`